src/internal.c

```diff
@@ -17709,9 +17709,11 @@ int SendData(WOLFSSL* ssl, const void* data, int sz)
 
         len = wolfSSL_GetMaxRecordSize(ssl, sz - sent);
 
-#ifdef WOLFSSL_DTLS
-        if (IsDtlsNotSctpMode(ssl)) {
-            len = min(len, MAX_UDP_SIZE);
+#if defined(WOLFSSL_DTLS) && !defined(WOLFSSL_NO_DTLS_SIZE_CHECK)
```

Comment:
Should we add a test case for `DTLS_SIZE_ERROR` with `!WOLFSSL_NO_DTLS_SIZE_CHECK`?

Reply:
Added a test case for this.